training-event/src/test/java/org/example/integration/steps/TrainingEventSteps.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TrainingEventSteps {
+    private final WebApplicationContext webApplicationContext;
+    private final ObjectMapper objectMapper;
+    @MockBean
+    private final TrainingService trainingEventService = Mockito.mock(TrainingService.class);
+
+    private MockMvc mockMvc;
+    private MvcResult mvcResult;
+
+    @Given("the training service is operational")
+    public void trainingServiceIsOperational() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext).build();
+
+        TrainingEventDto trainingEventDTO = new TrainingEventDto();
+        trainingEventDTO.setTrainerUsername("test");
+        trainingEventDTO.setTrainerFirstName("Test");
+        trainingEventDTO.setTrainerLastName("User");
+        trainingEventDTO.setActive(true);
+        trainingEventDTO.setTrainingDate(LocalDate.now());
+        trainingEventDTO.setTrainingDuration(200);
+        trainingEventDTO.setActionType(String.valueOf(ActionType.ADD));
+
+        trainingEventService.saveTrainingEvent(trainingEventDTO);
+    }
+
+    @Given("the training service is unavailable")
+    public void trainingServiceIsUnavailable() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext).build();
+
+        Mockito.doThrow(new RuntimeException("Service unavailable")).when(trainingEventService).countTrainerSummary();
+    }
+
+
+    @When("I send a GET request to {string}")
+    public void sendGetRequest(String url) throws Exception {
+        this.mvcResult = mockMvc.perform(MockMvcRequestBuilders.get(url))
+                .andExpect(status().isOk())
+                .andReturn();
+    }
+
+    @When("I send a GET request for unavailable service to {string}")
+    public void sendUnavailableGetRequest(String url) throws Exception {
+        TrainingEventDto trainingEventDTO = new TrainingEventDto();
+        trainingEventDTO.setTrainerUsername("test");
+        trainingEventDTO.setTrainerFirstName("Test");
+        trainingEventDTO.setTrainerLastName("User");
+        trainingEventDTO.setActive(true);
+
+        String requestBody = objectMapper.writeValueAsString(trainingEventDTO);
+
+        this.mvcResult = mockMvc.perform(MockMvcRequestBuilders.get(url)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestBody))
+                .andReturn();
```

Comment:
what makes this method for "unavalaible" call? How it different from "available" call? extract dto and use generic method "url {string} is requested"